packages/next/server/config.ts

```diff
@@ -615,6 +615,42 @@ function assignDefaults(dir: string, userConfig: { [key: string]: any }) {
     result.transpilePackages = (result.experimental as any).transpilePackages
   }
 
+  if (
+    result.experimental &&
+    'allowMiddlewareResponseBody' in (result.experimental as any)
+  ) {
+    Log.warn(
+      `\`allowMiddlewareResponseBody\` has been moved out of \`experimental\`. Please update your ${configFileName} file accordingly.`
+    )
+    result.allowMiddlewareResponseBody = (
+      result.experimental as any
+    ).allowMiddlewareResponseBody
+  }
+
+  if (
+    result.experimental &&
+    'skipMiddlewareUrlNormalize' in (result.experimental as any)
+  ) {
+    Log.warn(
+      `\`skipMiddlewareUrlNormalize\` has been moved out of \`experimental\`. Please update your ${configFileName} file accordingly.`
+    )
+    result.skipMiddlewareUrlNormalize = (
+      result.experimental as any
+    ).skipMiddlewareUrlNormalize
+  }
+
+  if (
+    result.experimental &&
+    'skipTrailingSlashRedirect' in (result.experimental as any)
+  ) {
+    Log.warn(
+      `\`skipTrailingSlashRedirect\` has been moved out of \`experimental\`. Please update your ${configFileName} file accordingly.`
+    )
+    result.skipTrailingSlashRedirect = (
+      result.experimental as any
+    ).skipTrailingSlashRedirect
+  }
```

Comment:
In the future, we might want to make a helper function to avoid the boilerplate (keep it DRY)